pkg/tenant/apikeys.go

```diff
@@ -160,6 +170,17 @@ func (s *Server) ProjectAPIKeyCreate(c *gin.Context) {
 		return
 	}
 
+	// Ensure that the user can't create an API key with permissions they don't have
+	// This is a lightweight check to prevent bad requests from going to Quarterdeck
+	for _, permission := range params.Permissions {
+		if perms.UserKeyPermission(permission) && !claims.HasPermission(permission) {
+			// Do not allow the user to create an apikey with this permission
+			sentry.Warn(c).Msg("user tried to create an API key with permissions they don't have")
+			c.JSON(http.StatusBadRequest, api.ErrorResponse("invalid permissions requested for API key"))
+			return
+		}
+	}
```

Comment:
Note that Quarterdeck also does this check (currently using the same method but eventually Quarterdeck could do a more robust check using the database).